remote/remote_test.go

```diff
@@ -168,3 +171,84 @@ func doTestWatch(t *testing.T, sm subnet.Manager) {
 		t.Errorf("WatchSubnet produced wrong subnet: expected %s, got %s", l.Key(), evt.Lease.Key())
 	}
 }
+
+func doTestWatchNetworks(t *testing.T, sm subnet.Manager, serverRegistry *subnet.MockSubnetRegistry) {
+	ctx, cancel := context.WithCancel(context.Background())
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	defer func() {
+		cancel()
+		wg.Wait()
+	}()
+
+	// Because remote clients are not allowed to add/remove networks, there
+	// is no server API for these operations.  But testing network watches
+	// still requires this, and that must happen directly in the server's
+	// registry.  That can cause race conditions if the WatchNetworks
+	// goroutine hasn't been run yet by the time CreateNetwork is called,
+	// so testing operations that touch the server registry must be deferred
+	// a short time to ensure that the server is listening when they happen.
+
+	events := make(chan []subnet.Event)
+	go func() {
+		subnet.WatchNetworks(ctx, sm, events)
+		wg.Done()
+	}()
+
+	// skip over the initial snapshot
+	<-events
+
+	expectedNetname := "foobar"
+	go func() {
+		// Add a new network a short time later to ensure that the
+		// client-side network watch triggers a server watch before the
+		// network is added directly in the server with CreateNetwork()
+		time.Sleep(time.Second)
```

Comment:
> I don't think I fully understand the need for sleep. Is the desire to exercise the code where the server does an etcd watch prior to the network key being created?

The server needs to be watching before the new network is created on the client side, otherwise the server-side will completely miss the event.  So subnet.WatchNetworks() must be running and sitting in RemoteManager::httpDo() (which triggers the server's handleWatchNetworks() function -> etcd.go::WatchNetworks()) before we execute the CreateNetwork() call.  But since subnet.WatchNetworks() is done in a goroutine and the actual HTTP request that triggers the server's handleWatchNetworks is way down the callstack, there's no way to guarantee that this will all happen because they are concurrent.

Obviously this is only a problem in the testcases, because normal clients will never be creating networks.

So possibly my approach of directly modifying the server's registry to add the network and generate the event isn't the right way to go about this, because of this problem...